### PR TITLE
Add 2545 natives.

### PR DESCRIPTION
**Pull request by 4mmonium**
This should import all 2545 natives, I made a tool to import them, I compared the markdown, I hope I didn't miss anything 😀

### CODE REVIEW

**Comment by blattersturm** [comment]
Why'd build fail here? Also, b2612 🙃

**Comment by 4mmonium** [comment]
> Is there any reason/purpose for `JOAAT_UNKNOWN`? Is the convention changing?

No, I thought it was a good idea since it makes it easier to find natives with missing joaat hashes (by searching JOAAT_UNKNOWN) later on and update them if needed.

**Comment by 4mmonium** [comment]
> > Is there any reason/purpose for `JOAAT_UNKNOWN`? Is the convention changing?
> 
> No, I thought it was a good idea since it makes it easier to find natives with missing joaat hashes (by searching `JOAAT_UNKNOWN`) later on and update them if needed. Done with the best intentions of course
> 
> For stuff like this for example: [d87e718](https://github.com/citizenfx/natives/commit/d87e718cc4a679ab5f175512a1cd7cc70e05c33b)
> 
> `JOAAT_UNKNOWN` only gets set when the `jhash` JSON field is empty. https://raw.githubusercontent.com/alloc8or/gta5-nativedb-data/master/natives.json

I corrected this, so instead of putting such in the native comment (potentially confusing users) I just placed a description for the native, telling the user the JOAAT hash is missing in case they want to fill it in.

**Old commit:** 
https://github.com/citizenfx/natives/commit/268b04141683d7d0f72cfaa4b8406ca788e5cb0f#diff-26795edb097dd19bf2a1bcc4f6e49e50a246c8e763f0378b8c698153055651caR8

**New commit:**
https://github.com/citizenfx/natives/pull/669/commits/8fca7b1a53ad4908ba2d2c2ecf9114fbc7a54579#diff-26795edb097dd19bf2a1bcc4f6e49e50a246c8e763f0378b8c698153055651caR12

**Comment by AvarianKnight** [comment]
It would be useless for most end users anyways, plus with how native docs work now this would probably also give them `Has "modern" docs` even though they dont

**Comment by 4mmonium** [comment]
> > > > Is there any reason/purpose for `JOAAT_UNKNOWN`? Is the convention changing?
> > > 
> > > 
> > > No, I thought it was a good idea since it makes it easier to find natives with missing joaat hashes (by searching `JOAAT_UNKNOWN`) later on and update them if needed. Done with the best intentions of course
> > > For stuff like this for example: [d87e718](https://github.com/citizenfx/natives/commit/d87e718cc4a679ab5f175512a1cd7cc70e05c33b)
> > > `JOAAT_UNKNOWN` only gets set when the `jhash` JSON field is empty. https://raw.githubusercontent.com/alloc8or/gta5-nativedb-data/master/natives.json
> > 
> > 
> > I corrected this, so instead of putting such in the native comment (potentially confusing users) I just placed a description for the native, telling the user the JOAAT hash is missing in case they want to fill it in.
> > **Old commit:** [268b041#diff-26795edb097dd19bf2a1bcc4f6e49e50a246c8e763f0378b8c698153055651caR8](https://github.com/citizenfx/natives/commit/268b04141683d7d0f72cfaa4b8406ca788e5cb0f#diff-26795edb097dd19bf2a1bcc4f6e49e50a246c8e763f0378b8c698153055651caR8)
> > **New commit:** [8fca7b1#diff-26795edb097dd19bf2a1bcc4f6e49e50a246c8e763f0378b8c698153055651caR12](https://github.com/citizenfx/natives/commit/8fca7b1a53ad4908ba2d2c2ecf9114fbc7a54579#diff-26795edb097dd19bf2a1bcc4f6e49e50a246c8e763f0378b8c698153055651caR12)
> 
> this is kinda pointless, since for "brand-new" natives there will probably be no chance to get your hands on the jooat hash ever, recently chipjpg found some missing ones which were shipped in one of the last pushed xbox 360 builds (Title Update 27, 2016/2017)

Yeah, I did this taking into consideration what chipjpg did, but I removed it now since it doesn't make any sense at all and as @AvarianKnight said, it will just cause doc natives to show `Has "modern" docs` even though they don't.

**Comment by gottfriedleibniz** [comment]
Is this ever going to be merged? 2545 was released ~18 months ago. Same w/ 2612 and 2802.

File: VEHICLE/SetHeliResistToExplosion.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+void _SET_HELI_RESIST_TO_EXPLOSION(Vehicle vehicle, BOOL p1);
+```
+
+```
+NativeDB Introduced: v2545
+```
+
+## Parameters
+* **vehicle**:
+* **p1**:
```
Comment (PsychoShock) [inline VEHICLE/SetHeliResistToExplosion.md:18]: P1 is if vehicle resist to explosion or not.

File: TASK/TaskWarpPedDirectlyIntoCover.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+NativeDB Introduced: v2545
+```
+
+## Parameters
+* **p0**:
+* **p1**:
+* **p2**:
+* **p3**:
+* **p4**:
+* **p5**:
```
Comment (PsychoShock) [inline TASK/TaskWarpPedDirectlyIntoCover.md:22]: * **ped**: The ped index.
* **canpeek**: If ped can peek and aim. False by default.
* **forceinitialdirection**: False by default.
* **forcefaceleft**: False by default. 
* **coverpoint**: Cover index.


I don't seem any `p5`.

File: STATS/PlaystatsInstMissionEnd.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+ns: STATS 
+---
+
+## _PLAYSTATS_INST_MISSION_END
+
+```c
+// 0xFEA3F7E83C0610FA 
+void _PLAYSTATS_INST_MISSION_END(Any p0);
+```
+
```
Comment (PsychoShock) [inline STATS/PlaystatsInstMissionEnd.md:11]: This native activatesthe  following completion of instanced missions, commencing with the Fixer pack. This functionality isn't intended for competitive modes tho.

File: STATS/PlaystatsHubExit.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+---
+ns: STATS 
+---
+
+## _PLAYSTATS_HUB_EXIT
+
+```c
+// 0x5A46ACE5C4661132 
+void _PLAYSTATS_HUB_EXIT(Any p0);
```
Comment (PsychoShock) [inline STATS/PlaystatsHubExit.md:9]: This is a list of the struct hub exit from what I found.

- `hubID`: Identifies this specific instance of the social hub, shared among players within it.
- `type`: Indicates the type of social hub the player departed from (e.g., music studio, smoking room).
- `bossId1` and `bossId2`: Unique identifiers for Boss instances present.
- `playerRole`: Denotes the player's role within the hub (e.g., CEO, bodyguard, biker prospect).
- `pulled`: Indicates if the player was forcibly removed from the social hub (e.g., due to a mission starting).
- `crewId`: Identifies the Social Club Crew to which the player belongs.
- `duration`: Records the time spent within the social hub.
- `dre`: Signifies the presence of Dr. Dre within the social hub.
- `playerCount`: Reflects the number of players present within the social hub at the time of exiting.

File: OBJECT/N_0x5CE2E45A5CE2E45A.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+---
+ns: OBJECT 
+---
+
+## _0x5CE2E45A5CE2E45A
+
+```c
+// 0x5CE2E45A5CE2E45A 
+void _0x5CE2E45A5CE2E45A(Any p0);
```
Comment (PsychoShock) [inline OBJECT/N_0x5CE2E45A5CE2E45A.md:9]: Name seems to be `FORCE_PORTABLE_PICKUP_LAST_ACCESSIBLE_POSITION_SETTING` with 1 param (The pickup id).

File: NETWORK/N_0xFEA7A352DDB34D52.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+---
+ns: NETWORK 
+---
+
+## _0xFEA7A352DDB34D52
+
+```c
+// 0xFEA7A352DDB34D52 
+void _0xFEA7A352DDB34D52(Any p0);
```
Comment (PsychoShock) [inline NETWORK/N_0xFEA7A352DDB34D52.md:9]: Seems to be named `NETWORK_BLOCK_PROXY_MIGRATION_BETWEEN_TUTORIAL_SESSIONS` with param BOOL if block or not. 

This native seems to prevents the transfer of network objects between players situated in different tutorial instances. I guess the flag needs to be deactivated when its purpose is fulfilled or no longer required.

File: NETWORK/N_0xC6FCEE21C6FCEE21.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
+---
+ns: NETWORK 
+---
+
+## _0xC6FCEE21C6FCEE21
+
+```c
+// 0xC6FCEE21C6FCEE21 
+void _0xC6FCEE21C6FCEE21(Any p0);
```
Comment (PsychoShock) [inline NETWORK/N_0xC6FCEE21C6FCEE21.md:9]: Seems to be named `NETWORK_SET_TASK_CUTSCENE_INSCOPE_MULTIPLER` with float param for multiplier value. I bet it alters the scoping distance of `CTaskCutscene` for remote players. R* seems to reset to 1.0 when no longer needed.

File: MONEY/NetworkSpentUpgradeAgency.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+```
+NativeDB Introduced: v2545
+```
+
+## Parameters
+* **p0**:
+* **p1**:
+* **p2**:
+* **p3**:
```
Comment (PsychoShock) [inline MONEY/NetworkSpentUpgradeAgency.md:20]: Param's seems to be BOOL `amount`, BOOL `if from bank AND wallet`, CHAR `context`, and CHAR `option` that seems to be 0 by default.

File: MONEY/NetworkSpentSuvFastTravel.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+```
+NativeDB Introduced: v2545
+```
+
+## Parameters
+* **p0**:
+* **p1**:
+* **p2**:
+* **p3**:
```
Comment (PsychoShock) [inline MONEY/NetworkSpentSuvFastTravel.md:20]: Param's seems to be BOOL `amount`, BOOL `if from bank AND wallet` and CHAR `npcprovider`,

File: MONEY/NetworkSpentSupplyContactService.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+```
+NativeDB Introduced: v2545
+```
+
+## Parameters
+* **p0**:
+* **p1**:
+* **p2**:
+* **p3**:
```
Comment (PsychoShock) [inline MONEY/NetworkSpentSupplyContactService.md:20]: Param's seems to be BOOL `amount`, BOOL `if from bank`, BOOL `if from bank AND wallet` and `npcprovider`.

File: HUD/SetCurrentCharacterHudColor.md [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+ns: HUD 
+---
+
+## _SET_CURRENT_CHARACTER_HUD_COLOR
+
+```c
+// 0x2ACCB195F3CCD9DE 
+void _SET_CURRENT_CHARACTER_HUD_COLOR(int hudColorId);
+```
+
```
Comment (PsychoShock) [inline HUD/SetCurrentCharacterHudColor.md:11]: This native enables the script to designate a personalized color for the multiplayer HUD, akin to modifying colors for characters in multiplayer. Need to clear this by passing -1 when no longer in use.

File: GRAPHICS/N_0xC6730E0D14E50703.md [hunk 12]
```diff
@@ -0,0 +1,17 @@
+---
+ns: GRAPHICS 
+---
+
+## _0xC6730E0D14E50703
+
+```c
+// 0xC6730E0D14E50703 
+void _0xC6730E0D14E50703(Any p0);
```
Comment (PsychoShock) [inline GRAPHICS/N_0xC6730E0D14E50703.md:9]: Correct name would be `SET_PTFX_FORCE_VEHICLE_INTERIOR_FLAG` with BOOL `value` to instructs all effect instances to set a flag, considering them as part of the vehicle interior.

**Review by PsychoShock — CHANGES_REQUESTED** [review]
Old but can be good to know if it get merged one day!

**Pull request closed without merging**